datasync/chaindatafetcher/kas/contract_caller_test.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kas
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/klaytn/klaytn/api"
+	"github.com/klaytn/klaytn/blockchain/vm"
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/common/hexutil"
+	"github.com/klaytn/klaytn/datasync/chaindatafetcher/kas/mocks"
+	"github.com/klaytn/klaytn/networks/rpc"
+	"github.com/stretchr/testify/suite"
+	"math/big"
+	"testing"
+	"time"
+)
+
+var (
+	decodedTrue         = hexutil.MustDecode("0x0000000000000000000000000000000000000000000000000000000000000001")
+	decodedFalse        = hexutil.MustDecode("0x0000000000000000000000000000000000000000000000000000000000000000")
+	ikip13Input         = hexutil.MustDecode("0x01ffc9a701ffc9a700000000000000000000000000000000000000000000000000000000")
+	ikip13InvalidInput  = hexutil.MustDecode("0x01ffc9a7ffffffff00000000000000000000000000000000000000000000000000000000")
+	ikip7Input          = hexutil.MustDecode("0x01ffc9a76578737100000000000000000000000000000000000000000000000000000000")
+	ikip7MetadataInput  = hexutil.MustDecode("0x01ffc9a7a219a02500000000000000000000000000000000000000000000000000000000")
+	ikip17Input         = hexutil.MustDecode("0x01ffc9a780ac58cd00000000000000000000000000000000000000000000000000000000")
+	ikip17MetadataInput = hexutil.MustDecode("0x01ffc9a75b5e139f00000000000000000000000000000000000000000000000000000000")
+)
+
+type SuiteContractCaller struct {
+	suite.Suite
+	ctrl   *gomock.Controller
+	api    *mocks.MockBlockchainAPI
+	caller *contractCaller
+}
+
+func (s *SuiteContractCaller) SetupTest() {
+	s.ctrl = gomock.NewController(s.T())
+	s.api = mocks.NewMockBlockchainAPI(s.ctrl)
+	s.caller = newContractCaller(s.api)
+}
+
+func (s *SuiteContractCaller) TearDownTest() {
+	s.ctrl.Finish()
+}
+
+func setExpectation(m *mocks.MockBlockchainAPI, contract *common.Address, blockNumber *big.Int, data, result []byte) {
+	arg := api.CallArgs{
+		From: common.Address{},
+		To:   contract,
+		Data: data,
+	}
+
+	m.EXPECT().Call(gomock.Any(), gomock.Eq(arg), gomock.Eq(rpc.BlockNumber(blockNumber.Int64()))).Return(result, nil).Times(1)
```

Comment:
So we don't check exact function but just check the arguments, right?

Reply:
We checks the function `Call` and its parameter, but not the first one which is a context.

Reply:
Oh, I misunderstood 😅